File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/model/EconomyOperation.java
package top.ourisland.creepersiarena.core.command.model;

/**
 * Mutating wallet operation used by /ciaa economy.
 */
public enum EconomyOperation {

    GIVE("give"),
    TAKE("take"),
    SET("set");

    private final String id;

    EconomyOperation(String id) {
        this.id = id;
    }

    public String id() {
        return id;
    }

}

File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/model/EconomyOperation.kt
package top.ourisland.creepersiarena.core.command.model

/**
 * Mutating wallet operation used by /ciaa economy.
 */
enum class EconomyOperation(
    private val rawId: String
) {

    GIVE("give"),
    TAKE("take"),
    SET("set");

    fun id(): String = rawId

}
